client/lib/interval/README.md

```diff
@@ -0,0 +1,43 @@
+# Interval
+
+An interface into a global timer coalescing interval runner.
+
+_**An interface**_ because this component handles registering and un-registering the given `onTick` action with the global runner.
+
+_**timer coalescing**_ because the global runner will run at the same time all of the actions registered for a given interval period. For example, if four actions are registered for running once per minute, they will all run at one point in time every minute instead of having four different run times, each time repeating every minute. This is better for things like battery life because it allows for longer and deeper periods of sleep between actions.
+
+_**interval runner**_ because the global runner will execute the given action every interval on the interval.
+
+This component can be used to easily trigger a polling, looping, or interval action in the background. Such usages could include polling an API endpoint for updates, sweeping over user input such as in the post editor at intervals, or updating an on-screen timer.
+
+It only requires two inputs: an action to perform and an interval between which runs of the action should occur. The action is simply a function and the interval is a named constant representing the interval period. These interval periods are intentionally limited to prevent sprawl of timers.
+
+The action will only be executed as long as the React component is mounted, as the component un-registers the action on unmount. Additionally, the default behavior is to stop executing the action when the browser document is hidden, though this can be overwritten.
+
```

Comment:
Mozilla says that modern browsers clamp to 1000ms in background tabs, so this _shouldn't_ impact us since that's the smallest interval anyway.

https://developer.mozilla.org/en-US/docs/Web/API/WindowTimers/setTimeout